packages/baseclient-components/src/container/LayerLegendAccordion/LayerLegendAccordion.tsx

```diff
@@ -0,0 +1,496 @@
+          placement="right"
+          mouseEnterDelay={0.5}
+        >
+          <span
+            className={visibilitySpanClass}
+            onClick={() => this.onLayerTreeNodeVisibilityChange(layer)}
+          />
+        </Tooltip>
+        <Tooltip
+          title={layer.get('name')}
+          placement="top"
+          mouseEnterDelay={0.5}
+        >
+          <span
+            className="layer-tree-node-title-layername"
+          >
+            {layer.get('name')}
+          </span>
+        </Tooltip>
+      </span>;
+    }
+
+    let loading;
+    if (layer &&
+      layer.getVisible() &&
+      (layer.getSource() instanceof OlImageWmsSource || layer.getSource() instanceof OlTileWmsSource)) {
+      const wmsSource: any = layer.getSource();
+      const encodedLayerNames = wmsSource.getParams().LAYERS;
+      loading = loadingQueue!.find((e: string) => e.indexOf(encodedLayerNames) > -1);
+    }
+    const loadingSpanClass = `fa fa-spinner fa-spin layer-tree-node-loading-${loading ? 'active' : 'inactive'}`;
+
+    return (
+      <span
+        className="layer-tree-node-list-item"
+      >
+        <div className="layer-tree-node-title">
+        <Tooltip
+          title={t('LayerLegendAccordion.toggleVisibilityTooltipText')}
+          placement="right"
+          mouseEnterDelay={0.5}
+        >
+          <span
+            className={visibilitySpanClass}
+            onClick={() => this.onLayerTreeNodeVisibilityChange(layer)}
+          />
+        </Tooltip>
+        <Tooltip
+          title={layer.get('name')}
+          placement="top"
+          mouseEnterDelay={0.5}
+        >
+          <span
+            className="layer-tree-node-title-layername"
+          >
+            {layer.get('name')}
+          </span>
+        </Tooltip>
+        <span
+            className={loadingSpanClass}
+          />
+        {
+          !(layer instanceof OlLayerGroup) ?
+            <Dropdown
+              overlay={<div />} // TODO: make something more senseful here
+              placement="topLeft"
+              trigger={['click']}
+            >
+              <Tooltip
+                title={this.props.t('LayerLegendAccordion.layerSettingsTooltipText')}
+                placement="right"
+                mouseEnterDelay={0.5}
+              >
+                <span
+                  className="fa fa-cog layer-tree-node-title-settings"
+                />
+              </Tooltip>
+            </Dropdown> :
+            null
+          }
+          </div>
+          <LayerTransparencySlider
+            layer={layer}
+          />
+        </span>
+    );
+  }
+
+  /**
+   * Generate legend items for currently visible layers on map
+   *
+   * @returns {Collapse[]} An array of collapsible Instances of <Legend>
+   */
+  getLegendItems() {
+    const { map } = this.props;
+    const layers = map.getLayerGroup().getLayers().getArray();
+    const scale = MapUtil.getScaleForResolution(map.getView().getResolution(), 'm');
```

Comment:
'm' can be replaced by map.getProjection().getUnits(), to be more compatible

Reply:
:+1: see 913024be0